crates/cxx-qt-lib/src/core/qmarginsf.rs

```diff
@@ -0,0 +1,101 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx::{type_id, ExternType};
+use std::fmt;
+
+#[cxx::bridge]
+mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qmargins.h");
+        type QMargins = crate::QMargins;
+        include!("cxx-qt-lib/qmarginsf.h");
+        type QMarginsF = super::QMarginsF;
+        include!("cxx-qt-lib/qstring.h");
+        type QString = crate::QString;
+
+        /// Returns true if all margins are very close to 0; otherwise returns false.
+        #[rust_name = "is_null"]
+        fn isNull(self: &QMarginsF) -> bool;
+
+        /// Returns an integer-based copy of this margins object.
+        ///
+        /// Note that the components in the returned margins will be rounded to the nearest integer.
+        #[rust_name = "to_margins"]
+        fn toMargins(self: &QMarginsF) -> QMargins;
+    }
+
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/common.h");
+
+        #[doc(hidden)]
+        #[rust_name = "qmarginsf_default"]
+        fn construct() -> QMarginsF;
+        #[doc(hidden)]
+        #[rust_name = "qmarginsf_from_qmargin"]
+        fn construct(margins: &QMargins) -> QMarginsF;
+        #[doc(hidden)]
+        #[rust_name = "qmarginsf_new"]
+        fn construct(left: f64, top: f64, right: f64, bottom: f64) -> QMarginsF;
+        #[doc(hidden)]
+        #[rust_name = "qmarginsf_to_qstring"]
+        fn toQString(value: &QMarginsF) -> QString;
+    }
+}
+
+/// The QMarginsF class defines the four margins of a rectangle.
+#[derive(Debug, Clone, PartialEq)]
+#[repr(C)]
+pub struct QMarginsF {
```

Comment:
Hm, I'd just like to double-check whether it makes sense to expose these members as `pub` :thinking: 

It definitely leads to more rust-y code.
However, these members are not public in Qt's C++ code.
So could a new Qt minor version change the meaning of these members without breaking backwards compatibility?
I think not, because the types must be binary-compatible, but unsure :thinking: 
As the members aren't exposed, they would also have to be constructed by Qt code, which could change whatever each field means.
e.g. change it to `left`, `bottom`, `widht` and `height`. which wouldn't change the binary representation.

So do we want to risk this happening? It's not very likely though I'd guess.

Reply:
Right, maybe for now we just map the Qt public API across and not the internals ? And do the same for methods that are weird in Qt (eg keep the Qt API for now for things like QRect). Then in a future cycle we can look at exposing the internal fields and improving the methods ?

We also need to be careful that in some types we use derive, but i think for Eq / Ord it should be mainly OK ?

Reply:
done